django/core/paginator.py

```diff
@@ -236,3 +359,72 @@ def end_index(self):
         if self.number == self.paginator.num_pages:
             return self.paginator.count
         return self.number * self.paginator.per_page
+
+
+class AsyncPage:
+    def __init__(self, object_list, number, paginator):
+        self.object_list = object_list
+        self.number = number
+        self.paginator = paginator
+
+    def __repr__(self):
+        return "<Async Page %s>" % self.number
+
+    async def __aiter__(self):
+        await self._afetch_object_list()
```

Comment:
how do you feel about not calling `_afetch_object_list` and doing `async for` explicitally?

```suggestion
    if hasattr(self.object_list, "__aiter__"):  # object_list is a queryset
        async for obj in self.object_list:
            yield obj
    else:
        for obj in self.object_list:
            yield obj
```

the reason being, if the implementation of queryset changes in the future to be more async aware, no re-write would be necessary

Reply:
Hi
If `self.object` is a queryset, wouldn't it hit the database every time `__aiter__` is called? 🤔